### ISSUE TITLE
Not able to use the nmt.translate.py script with a local GPU

### ISSUE THREAD

**Issue by davidbaines** [sillsdev/silnlp#139]
This error seems to be related to ClearML while trying to run the translate script locally.  
The error is different than that in #120 but the umbrella issue is not being able to use the pipeline with a local GPU.
```

(silnlp-gt_VMn9E-py3.8) david@pop-os:~/silnlp$ python -m silnlp.nmt.translate --checkpoint last --src-project tl-TCB --src /home/david/disk2/gutenberg/Paratext/projects/TCB/091SAtlASD15.SFM --trg-iso blx --output-usfm /home/david/disk2/gutenberg/BT-Tagalog/to_blx/tl_blx_uni_dup_share_preserve/results/091SAAMIU_last.sfm BT-Tagalog/to_blx/tl_blx_uni_dup_share_preserve
2022-07-16 15:03:40,452 - silnlp.common.environment - INFO - Using workspace: /home/david/disk2/gutenberg as per environment variable SIL_NLP_DATA_PATH.
2022-07-16 15:03:40,828 - silnlp.common.utils - INFO - Git commit: 8cd5b9c649
Traceback (most recent call last):
  File "/usr/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/david/silnlp/silnlp/nmt/translate.py", line 231, in <module>
    main()
  File "/home/david/silnlp/silnlp/nmt/translate.py", line 225, in main
    translator.translate_text_file(args.src, args.trg_iso, args.trg)
  File "/home/david/silnlp/silnlp/nmt/translate.py", line 151, in translate_text_file
    self.init_translation_task(experiment_suffix=f"_{self.checkpoint}_{os.path.basename(src_file_path)}")
  File "/home/david/silnlp/silnlp/nmt/translate.py", line 79, in init_translation_task
    self.clearml = SILClearML(
  File "<string>", line 9, in __init__
  File "/home/david/silnlp/silnlp/common/clearml_connection.py", line 24, in __post_init__
    self.name = self.name.replace("\\", "/")
AttributeError: 'NoneType' object has no attribute 'replace'
```

**Comment by davidbaines** [comment]
I now have a different error than the one above.  After the usual information about the model the error message is: 
`Was not able to translate ACT.  Error: 2`
[issue139.txt](https://github.com/sillsdev/silnlp/files/9778669/issue139.txt)

**Comment by davidbaines** [comment]
Investigating this with Michael and the problem may be related to the inclusion of the `--trg-iso en` argument on the command line. 
When running without that argument, on a local machine, the code did the inferencing without an error.  
However on ClearML running with the argument  `--trg-iso en` didn't report an error - but neither did it overwrite the existing translation. 

I'm certain that I have run this on a local machine with that argument and without an error in the past.

**Comment by johnml1135** [comment]
In taking a look at the error (and translate.py), it appears that it needs an experiment name and that none may be given.  Am I reading that correctly?

**Comment by davidbaines** [comment]
Yes, it does require an experiment name however I'm getting the same error
when I have the experiment name specified correctly and also have
`--trg-iso en` as an argument.  I do get the same error if there is an
experiment specified but it has a spelling error in the path. In that case
it's the same as if a non-existent experiment is specified.  It would be
great to have a more informative error message in that case.

**Comment by johnml1135** [comment]
Hello David - I think I found it.  You are using an older version of the code - you need to update to the most recent master.  There was a error put into translate.py in July adding an __init__() function call.  This is normally fine, but when using \@dataclass, python overrides the __init__ function and you should use __post_init__ instead.  Damien fixed this in August on commit cb8ba71.  If you update to the most recent master, this includes the fix and should resolve the "name" issue.

![image](https://user-images.githubusercontent.com/13733556/196571986-3927d7ab-434a-42b7-a55b-5267e6936d7d.png)
